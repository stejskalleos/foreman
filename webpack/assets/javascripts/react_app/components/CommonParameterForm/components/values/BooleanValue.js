import React from 'react';
import PropTypes from 'prop-types';

import { translate as __ } from '../../../../common/I18n';

const BooleanValue = ({ id, name, css, value, onChange }) => {
  const options = [
    { label: '', value: '' },
    { label: __('Yes'), value: 'true' },
    { label: __('No'), value: 'false' },
  ];
  return (
    <select
      id={id}
      name={name}
      className={`without_select2 ${css}`}
      onChange={e => onChange(e.target.value)}
      defaultValue={value}
    >
      {options.map(item => (
        <option value={item.value} key={item.value}>
          {item.label}
        </option>
      ))}
    </select>
  );
};

BooleanValue.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  css: PropTypes.string,
  value: PropTypes.any,
  onChange: PropTypes.func.isRequired,
};

BooleanValue.defaultProps = {
  id: '',
  css: '',
  value: false,
};

export default BooleanValue;
