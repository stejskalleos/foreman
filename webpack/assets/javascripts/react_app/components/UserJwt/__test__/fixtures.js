import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
