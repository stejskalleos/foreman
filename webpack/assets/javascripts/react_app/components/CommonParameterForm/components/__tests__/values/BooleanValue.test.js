import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { booleanValueFixtures } from '../fixtures';
import BooleanValue from '../../values/BooleanValue';


describe('BooleanValue', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(BooleanValue, booleanValueFixtures));
});
