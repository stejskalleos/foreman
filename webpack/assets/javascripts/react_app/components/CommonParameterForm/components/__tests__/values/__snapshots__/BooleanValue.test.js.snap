// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanValue rendering when value = "" 1`] = `
<select
  className="without_select2 form-control"
  defaultValue=""
  id="common_parameter_value"
  name="common_parameter[value]"
  onChange={[Function]}
>
  <option
    key=""
    value=""
  />
  <option
    key="true"
    value="true"
  >
    Yes
  </option>
  <option
    key="false"
    value="false"
  >
    No
  </option>
</select>
`;

exports[`BooleanValue rendering when value = false 1`] = `
<select
  className="without_select2 form-control"
  defaultValue={false}
  id="common_parameter_value"
  name="common_parameter[value]"
  onChange={[Function]}
>
  <option
    key=""
    value=""
  />
  <option
    key="true"
    value="true"
  >
    Yes
  </option>
  <option
    key="false"
    value="false"
  >
    No
  </option>
</select>
`;

exports[`BooleanValue rendering when value = true 1`] = `
<select
  className="without_select2 form-control"
  defaultValue={true}
  id="common_parameter_value"
  name="common_parameter[value]"
  onChange={[Function]}
>
  <option
    key=""
    value=""
  />
  <option
    key="true"
    value="true"
  >
    Yes
  </option>
  <option
    key="false"
    value="false"
  >
    No
  </option>
</select>
`;
